multiaddr_test.go

```diff
@@ -166,6 +166,7 @@ func TestBytesToString(t *testing.T) {
 	testString("/ip4/127.0.0.1/udp/1234", "047f0000011104d2")
 	testString("/ip4/127.0.0.1/tcp/4321", "047f0000010610e1")
 	testString("/ip4/127.0.0.1/udp/1234/ip4/127.0.0.1/tcp/4321", "047f0000011104d2047f0000010610e1")
+	testString("/onion/aaimaq4ygg2iegci:80", "bc030010c0439831b48218480050")
```

Comment:
- is it ALWAYS tcp? where can we get confirmation of this? 
- even if it IS always is TCP today, does it make sense to list it so that we can plug other transports in over tor? (sctp, quic, etc)
- this is external-facing, right? the other side gets the port too? 
- if so, does this count as leaking identifying information? port numbers could be correlated. most onion sites are `:80` (afaik), and port numbers are hard, but _still_. it is _some_ info. not sure how much worse than response timing ...

could really use others' viewpoints on this. cc @david415 @leif